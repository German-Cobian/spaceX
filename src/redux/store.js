import { createStore, combineReducers, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';
import logger from 'redux-logger';
import missionsReducer from './missionsReducer';

const reducer = combineReducers({
  missionsReducer,
  // additional reducers could be added here
});

const store = createStore(
  reducer,
  applyMiddleware(logger, thunk),
);

export default store;
