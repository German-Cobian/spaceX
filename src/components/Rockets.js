/* eslint-disable camelcase */
import React from 'react';
import PropTypes from 'prop-types';
import { useDispatch } from 'react-redux';
import { Badge } from 'react-bootstrap';
import { setReserved, setCancelation } from '../redux/rocketsReducer';

const Rockets = (props) => {
  const {
    id, flickr_images, rocket_name, description, reserved,
  } = props;

  const dispatch = useDispatch();

  const newReservation = () => {
    dispatch(setReserved(id));
  };

  const cancelReservation = () => {
    dispatch(setCancelation(id));
  };

  return (
    <div key={id} className="d-flex flex-row my-4 mx-4">
      <div>
        <img src={flickr_images} alt="flickr_images" className="rockets-images" />
      </div>
      <div className="d-flex flex-column align-items-start">
        <h5 className="mx-5">{rocket_name}</h5>
        <p className="text-start mx-5">
          <Badge bg={reserved === true ? 'info' : 'bg-hide'}>{ reserved === true ? 'Reserved' : ''}</Badge>
          {description}
        </p>
        <button
          type="button"
          className={reserved === true ? 'btn btn-dark mx-5' : 'btn btn-primary mx-5'}
          onClick={reserved === true ? cancelReservation : newReservation}
        >
          {reserved === true ? 'Cancel Reservation' : 'Reserve Rocket'}
        </button>
      </div>
    </div>
  );
};

Rockets.propTypes = {
  id: PropTypes.number.isRequired,
  rocket_name: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
  flickr_images: PropTypes.string.isRequired,
  reserved: PropTypes.bool.isRequired,
};

export default Rockets;
