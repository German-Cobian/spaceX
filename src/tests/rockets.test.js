import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { render } from '@testing-library/react';
import Rockets from '../components/Rockets';
import '@testing-library/jest-dom/extend-expect';
import store from '../redux/store';
import RocketsMock from './mocks/RocketsMock';

describe('rocket component functionality', () => {
  test('matches Rockets component snapshot', () => {
    const tree = renderer.create(
      <Provider store={store}>
        <Rockets />
      </Provider>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('should render Rockets component correctly', () => {
    const component = render(
      <RocketsMock
        id="TestId"
        rocket_name="rocket one"
        description="description one"
        reserved={false}
      />,
    );
    expect(component.container).toHaveTextContent('rocket one');
    expect(component.container).toHaveTextContent('description one');
  });

  test('If reserved = true innertext = Cancel Reservation', () => {
    const component = render(
      <RocketsMock
        id="TestId"
        rocket_name="rocket two"
        description="description one"
        reserved
      />,
    );
    expect(component.container).toHaveTextContent('Cancel Reservation');
  });

  test('If reserved = false innertext = Reserve Rocket', () => {
    const component = render(
      <RocketsMock
        id="TestId"
        rocket_name="rocket two"
        description="description one"
        reserved={false}
      />,
    );
    expect(component.container).toHaveTextContent('Reserve Rocket');
  });
});
