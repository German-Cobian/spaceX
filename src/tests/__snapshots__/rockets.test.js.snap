// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rocket component functionality matches Rockets component snapshot 1`] = `
<div
  className="d-flex flex-row my-4 mx-4"
>
  <div>
    <img
      alt="flickr_images"
      className="rockets-images"
    />
  </div>
  <div
    className="d-flex flex-column align-items-start"
  >
    <h5
      className="mx-5"
    />
    <p
      className="text-start mx-5"
    >
      <span
        className="badge bg-bg-hide"
      >
        
      </span>
    </p>
    <button
      className="btn btn-primary mx-5"
      onClick={[Function]}
      type="button"
    >
      Reserve Rocket
    </button>
  </div>
</div>
`;
